### PR TITLE
[vLLM] Add continuous batching test

### LINKED ISSUE #5813: [vLLM] Add test/demo for continuous batching
vLLM plugin scheduler supports continuous batching; add a demo or test to verify this behaviour.

- If new prompts arrive while older requests are already decoding, the scheduler prioritizes prefill for the new prompts. If the current requests are less than batch_size (max_num_seqs).
- On the next decode-only step, decode runs for both old and newly prefilled requests.

### Verification:
- [ ] Verify that kv cache is updated properly.
- [ ] Verify the generated output for each prompt. 

Try single device and multi device scenarios.

### PR DESCRIPTION
**Pull request by alinakhanTT**
### Ticket
closes #5813

### What's changed
Adds an on device continuous batching test. Two prompts start generating, and once they are decoding two more are submitted, so their prefill joins a live decode batch.

Every output is compared token for token against a reference pass where each prompt runs alone. That is what catches a request whose KV was clobbered by the joiner, or a joiner reading stale KV from its slot, since both keep producing fluent text that a coherence check would happily accept. The test also asserts the two waves really overlapped, so it cannot quietly turn into an ordinary batched run.

Covers single device, plus llmbox under TP and DP+TP. All three pass locally on device, with every output identical to its reference.

The scheduler half of the issue (prefill first on arrival, then decode for old and new together) is already covered on CPU by test_ascend_scheduler_batching.py, so this fills in the device side.

### Checklist
- [X] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/integrations/vllm_plugin/generative/test_continuous_batching.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    ):
+        label = "A" if i < len(WAVE_A) else "B"
+        print(f"  wave {label} req {i}: {prompt!r} -> {text!r} (ref {ref_text!r})")
+        if expected.lower() not in ref_text.lower():
+            failures.append(
+                f"req {i} ({prompt!r}): reference itself is wrong, expected "
+                f"{expected!r}, got {ref_text!r} -- not a batching failure"
+            )
+            continue
+        if ids != ref_ids:
```
Comment (kmabeeTT) [inline tests/integrations/vllm_plugin/generative/test_continuous_batching.py:182]: `ids != ref_ids` compares two separate `generate()` calls, which is exactly the property #5520 says is not guaranteed under TP: the cross-chip lm_head reduction is not bit-reproducible, so a near-tie argmax can flip between otherwise-identical calls. `cpu_sampling=True` does not cover it — #5520 is in the reduction producing the logits, not in the sampler op. The conftest you import from says the same thing at line 151, where `assert_batch_grounded` is deliberately "tolerant of the #5520 near-tie fp drift" on these same four prompts.

Suggest keeping the exact-token oracle for `single_device` (no cross-chip reduction, so #5520 does not apply) and dropping the TP / DP+TP variants back to `_assert_overlapped` plus a grounded check. That still catches the failure you are actually hunting — a joiner clobbering a neighbour's KV shows up as garbage or a repeat-loop, not as a one-token near-tie flip — without asserting bit-reproducibility the stack does not currently provide. Alternatively keep it and xfail the llmbox variants against #5520 so the flake is attributed when it lands.

File: tests/integrations/vllm_plugin/generative/test_continuous_batching.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+MAX_NUM_SEQS = 4
+MAX_MODEL_LEN = 64
+# Wave A must outlive B's arrival, so it generates for longer.
+MAX_TOKENS_A = 24
+MAX_TOKENS_B = 12
+DECODE_TOKENS_BEFORE_ARRIVAL = 4
+# The first request pays engine warmup and graph compilation.
+ARRIVAL_TIMEOUT_S = 900
+
+WAVE_A = GROUNDED_BATCH_CHECKS[:2]
```
Comment (kmabeeTT) [inline tests/integrations/vllm_plugin/generative/test_continuous_batching.py:32]: `WAVE_A`/`WAVE_B` slice a shared constant at a fixed index, and the test only works while `len(GROUNDED_BATCH_CHECKS) <= MAX_NUM_SEQS`. Today it is exactly 4 == 4, so both waves fit concurrently. If anyone appends a fifth check to `GROUNDED_BATCH_CHECKS` — it is shared with `assert_batch_grounded` and its other callers — `WAVE_B` becomes 3, the total exceeds `max_num_seqs`, and one B request queues instead of joining the live decode batch. The test keeps passing while no longer testing continuous batching.

One line at module scope pins it:

    assert len(WAVE_A) + len(WAVE_B) <= MAX_NUM_SEQS, "waves must fit in one batch"

File: tests/integrations/vllm_plugin/generative/test_continuous_batching.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    ]
+
+    gate_task = asyncio.create_task(gate.wait())
+    done, _pending = await asyncio.wait(
+        {gate_task, *a_tasks},
+        timeout=ARRIVAL_TIMEOUT_S,
+        return_when=asyncio.FIRST_COMPLETED,
+    )
+    if not gate_task.done():
+        gate_task.cancel()
```
Comment (kmabeeTT) [inline tests/integrations/vllm_plugin/generative/test_continuous_batching.py:120]: The timeout path cancels `gate_task` but leaves `a_tasks` pending, so the `AssertionError` propagates while wave A is still generating into an engine that `_collect`'s `finally` is about to shut down. That yields "Task was destroyed but it is pending" noise on top of the real failure, right when the output is being read to diagnose a timeout. Cancelling `a_tasks` alongside `gate_task` keeps the failure message clean.

**Review by kmabeeTT — CHANGES_REQUESTED** [review]
Looks good overall, few things to consider changing

File: tests/integrations/vllm_plugin/generative/test_continuous_batching.py (hunk 2 above)
Comment (alinakhanTT) [reply tests/integrations/vllm_plugin/generative/test_continuous_batching.py:32]: Added!

File: tests/integrations/vllm_plugin/generative/test_continuous_batching.py (hunk 1 above)
Comment (alinakhanTT) [reply tests/integrations/vllm_plugin/generative/test_continuous_batching.py:182]: TP / DP+TP variants dropped back to _assert_overlapped plus a grounded check and exact-token oracle for single_device kept

File: tests/integrations/vllm_plugin/generative/test_continuous_batching.py (hunk 3 above)
Comment (alinakhanTT) [reply tests/integrations/vllm_plugin/generative/test_continuous_batching.py:120]: Done

**Review by kmabeeTT — APPROVED** [review]
Thanks for changes!
